src/SSHLibrary/library.py

```diff
@@ -832,9 +833,16 @@ def login_with_public_key(self, username, keyfile, password='',
         | Login With Public Key | johndoe          | /home/johndoe/.ssh/id_dsa | keyringpasswd |
 
         Argument `delay` was added in SSHLibrary 2.0.
+
+        To provide SSH agent support, `allow_agent` and `look_for_keys` were introduced in
+        SSHLibrary 3.0.0.
+        `allow_agent` enables the connection to the SSH agent.
+        `look_for_keys` enables the searching for discoverable private key files in `~/.ssh/`.
+        `allow_agent` and `look_for_keys` were not available prior to SSHLibrary 3.0.0 and are
+         ignored by Jython.
```

Comment:
Various things that could be fine-tuned:

- New arguments should be explained already earlier after existing `username`, `keyfile`, etc. are explained.

- Would be nice to have an example using these new options unless it gets very complicated.

- No need to mention "SSHLibrary 3.0.0" multiple times. Could have something like this at the end of the documentation instead:

  > Argument `delay` is new in SSHLibrary 2.0. Arguments `look_for_keys` and `allow_agend` are new in SSHLibrary 3.0 and do not work when using Jython.

Reply:
Points 1 and 3 are done. I'm not sure about the example though. Those arguments are directly passed paramiko, that handles the connection to the ssh agent and the login through it. Also, some configuration must be done manually by the users, on their machine like adding the keys to their, ~/.ssh directory. Other than setting these values to `True`, there isn't much that can be done through Robot.